TaskHouseUnitTests/WorkersControllerUnitTests.cs

```diff
@@ -0,0 +1,177 @@
+
+        [Fact]
+        public async void WorkerController_Get_ReturnsAllElementsInRepo_WhenGivenNoParameters()
+        {
+            repo = new FakeWorkerRepository();
+
+            IEnumerable<Worker> result = await controller.Get();
+
+            Assert.Equal(3, result.Count());
+        }
+
+        [Fact]
+        public async void WorkerController_Create_ReturnsObjectResult_withValid()
+        {
+            Worker TestWorker =  new Worker()
+            {
+                Id = 5,
+                Username = "Tusername",
+                Password = "+z490sXHo5u0qsSaxbBqEk9KsJtGqNhD8I8mVBdDJls=", //1234
+                Email = "test@test.com",
+                FirstName = "Bob7",
+                LastName = "Bobsen6",
+                Salt = "upYKQSsrlub5JAID61/6pA=="
+            };
+
+            await controller.Create(TestWorker);
+            var result = await controller.Get(TestWorker.Id);
+            var resultObjectResult = await controller.Get(TestWorker.Id) as ObjectResult;
+            var resultObject = resultObjectResult.Value as TaskHouseApi.Model.Worker;
+
+            Assert.IsType<ObjectResult>(result);
+            Assert.Equal(TestWorker.Id, resultObject.Id);
+        }
+        
+        [Fact]
+         public async void WorekrController_Create_ReturnsBadRequest_WhenGivenNullWorker()
+        { 
+            Worker worker = null;
+
+            var result = await controller.Create(worker);
+
+            Assert.IsType<BadRequestObjectResult>(result);
+          
+
+        }
+
+        [Fact]
+        public async void WorkerController_Update_ReturnsBadRequestResult_WhenIdIsInvalidAndWorkerIsNull()
+        { 
+            Worker worker = null;
+            int id = 2600;
+
+            var result = await controller.Update(id, worker);
+
+            Assert.IsType<BadRequestResult>(result); 
+        }
+
+        [Fact]
+        public async void  WorkerController_Delete_ActuallyDeletes_WhenIdIsValid()
+        {
+            Worker TestWorker =  new Worker()
+            {
+                Id = 5,
+                Username = "Tusername",
+                Password = "+z490sXHo5u0qsSaxbBqEk9KsJtGqNhD8I8mVBdDJls=", //1234
+                Email = "test@test.com",
+                FirstName = "Bob7",
+                LastName = "Bobsen6",
+                Salt = "upYKQSsrlub5JAID61/6pA=="
+            };
+
+            await controller.Delete(TestWorker.Id) ;
+            var result = await controller.Get(TestWorker.Id);
+
+            Assert.IsType<NotFoundResult>(result);
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+        [Fact] 
+        public async void WorkerController_Delete_ReturnsNotFoundResult_WhenIdIsInvalid()
+        { 
+            int id = 2500; 
+
+            var result = await controller.Delete(id);
+
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+
+        [Fact]
+        public async void WorkerController_Update_ReturnsActionResult_withValidId()
```

Comment:
Update has a lot of different scenarios, so there should be more than 2 tests for it